src/Microsoft.Extensions.Configuration.Binder/ConfigurationBinder.cs

```diff
@@ -79,49 +88,49 @@ public static void Bind(this IConfiguration configuration, object instance)
         }
 
         /// <summary>
         /// Extracts the value with the specified key and converts it to type T.
         /// </summary>
         /// <typeparam name="T">The type to convert the value to.</typeparam>
         /// <param name="configuration">The configuration.</param>
-        /// <param name="key">The configuration key for the value to convert.</param>
+        /// <param name="sectionKey">The key of the configuration section's value to convert.</param>
         /// <param name="defaultValue">The default value to use if no value is found.</param>
         /// <returns>The converted value.</returns>
-        public static T GetValue<T>(this IConfiguration configuration, string key, T defaultValue)
+        public static T GetValue<T>(this IConfiguration configuration, string sectionKey, T defaultValue)
         {
-            return (T)GetValue(configuration, typeof(T), key, defaultValue);
+            return (T)GetValue(configuration, typeof(T), sectionKey, defaultValue);
         }
 
         /// <summary>
         /// Extracts the value with the specified key and converts it to the specified type.
         /// </summary>
         /// <param name="configuration">The configuration.</param>
         /// <param name="type">The type to convert the value to.</param>
-        /// <param name="key">The configuration key for the value to convert.</param>
+        /// <param name="sectionKey">The key of the configuration section's value to convert.</param>
         /// <returns>The converted value.</returns>
-        public static object GetValue(this IConfiguration configuration, Type type, string key)
+        public static object GetValue(this IConfiguration configuration, Type type, string sectionKey)
         {
-            return GetValue(configuration, type, key, defaultValue: null);
+            return GetValue(configuration, type, sectionKey, defaultValue: null);
         }
 
         /// <summary>
         /// Extracts the value with the specified key and converts it to the specified type.
         /// </summary>
         /// <param name="configuration">The configuration.</param>
         /// <param name="type">The type to convert the value to.</param>
-        /// <param name="key">The configuration key for the value to convert.</param>
+        /// <param name="sectionKey">The key of the configuration section's value to convert.</param>
         /// <param name="defaultValue">The default value to use if no value is found.</param>
         /// <returns>The converted value.</returns>
-        public static object GetValue(this IConfiguration configuration, Type type, string key, object defaultValue)
+        public static object GetValue(this IConfiguration configuration, Type type, string sectionKey, object defaultValue)
         {
-            var value = configuration.GetSection(key).Value;
+            var value = configuration.GetSection(sectionKey).Value;
```

Comment:
Please rollback all the `sectionKey` to just `key`.

 That is an implementation detail. We could use `configuration[key]`. 

In `Bind` the purpose of the argument is to bind the object to the entire section (i.e. all subkeys) identified by the key. That is why I thought `sectionKey` made it clearer, but the value is minimal.  For `GetValue` overloads the purpose is to just get the value at the key. We don't care about subkeys so talking about section in this context does not help.